rust/main/chains/hyperlane-ethereum/src/contracts/mailbox.rs

```diff
@@ -683,10 +683,14 @@ where
 
     async fn process_calldata(
         &self,
-        _message: &HyperlaneMessage,
-        _metadata: &[u8],
+        message: &HyperlaneMessage,
+        metadata: &[u8],
     ) -> ChainResult<Vec<u8>> {
-        todo!()
+        let contract_call = self
+            .process_contract_call(message, metadata, None, false)
```

Comment:
I think `process_contract_call` already does all the estimations / network calls we're looking to decouple from the MessageProcessor. This should likely just get the plain ContractCall, with no estimates set in the TypedTransaction
```suggestion
        let contract_call = self
            .contract_call(message, metadata, None)
```

Reply:
actually even simpler:
```
        self.contract.process(
            metadata.to_vec().into(),
            RawHyperlaneMessage::from(message).to_vec().into(),
        );
```

Reply:
Yes, it should work.

Reply:
Done